subsys/dfu/dfu_target/src/dfu_stream_flatten.c

```diff
@@ -22,7 +22,7 @@ int stream_flash_flatten_page(struct stream_flash_ctx *ctx, off_t off)
 	}
 
 #if defined(CONFIG_STREAM_FLASH_ERASE)
-	if (ctx->last_erased_page_start_offset == page.start_offset) {
+	if (ctx->erased_up_to == page.start_offset) {
```

Comment:
This is not correct because `erased_up_to` is last erased offset so it should rather be:
```suggestion
	if (ctx->erased_up_to >= (page.start_offset + page.size)) {
```
and still that would not be correct, because the `erased_up_to` tracks how far Stream Flash tried to prepare device for write. It basically means that some time a go a range [0, erased_up_to] has been prepared for write, but it is unknown whether any writes happened at the same range, which is tracked by `bytes_written`, independently.

So the logic here will exit thinking that the page it tries to scramble is just erased (empty, not written), but that is not the case.